operator/cmd/k8s_identity.go

```diff
@@ -91,157 +221,117 @@ func identityGCIteration(ctx context.Context, clientset k8sClient.Clientset) {
 		return
 	}
 
-	identityStoreList := identityStore.List()
-	totalEntries := len(identityStoreList)
+	identities, err := igc.clientset.CiliumV2().CiliumIdentities().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		igc.logger.WithError(err).Error("unable to list Cilium identities")
+	}
+
+	totalEntries := len(identities.Items)
 	deletedEntries := 0
 
 	timeNow := time.Now()
-	for _, identityObject := range identityStoreList {
-		identity, ok := identityObject.(*v2.CiliumIdentity)
-		if !ok {
-			log.WithField(logfields.Object, identityObject).
-				Errorf("Saw %T object while expecting k8s/types.Identity", identityObject)
-			continue
-		}
-
+	for _, identity := range identities.Items {
 		// The identity is definitely alive if there's a CE using it.
 		if watchers.HasCEWithIdentity(identity.Name) {
```

Comment:
Unfortunate that we're still depending on this. Perhaps worth as next step to bring in `CiliumEndpoints` as shared resource.